packages/blade/src/components/Typography/BaseText/StyledBaseText.native.tsx

```diff
@@ -0,0 +1,35 @@
+import styled from 'styled-components/native';
+import getBaseTextStyles from './getBaseTextStyles';
+import type { StyledBaseTextProps } from './StyledBaseText.d';
+
+export const ERROR_AS_PROP_NOT_SUPPORTED = `[Blade:BaseText]: "as" prop is not supported for BaseText for React Native platform`;
+const StyledBaseText = styled.Text<StyledBaseTextProps>(
+  ({
+    color,
+    fontFamily,
+    fontSize,
+    fontWeight,
+    fontStyle,
+    textDecorationLine,
+    lineHeight,
+    textAlign,
+    as,
+  }) => {
+    if (as) {
+      throw new Error(ERROR_AS_PROP_NOT_SUPPORTED);
```

Comment:
Lets not use a const here and export that const to be re-used in tests. 

If I change this const's value into something non-meaningful, our tests will never catch it. Let's hardcode the test to ensure the right error is being thrown.